chapter/background.md

```diff
@@ -0,0 +1,607 @@
+
+Yet, despite these formats being simple to parse and widely supported [@json], their static nature rules out any dynamic content such as generated fields, functions and the possibility to factorize and reuse.
+Moreover, content validation has to be developed separately, which led to the design of complementary schema specification languages like json-schema [@json-schema] or XSD [@xsd].
+
+These qualities require an evaluated language.
+In fact, some applications make heavy use of config files written in the native programming language which gives them access to language features and existing analysis tools.
+Examples include JavaScript frameworks such as webpack [@webpack] or Vue [@vue] and python package management using `setuptools`[@setuptools].
+
+Despite this, not all languages serve as a configuration language, e.g. compiled languages and some domains require language agnostic formats.
+For particularly complex products, both language independence and advanced features are desirable.
+Alternatively to generating configurations using high level languages, this demand is addressed by more domain specific languages.
+Dhall [@dhall], Cue [@cue] or jsonnet [@jsonnet] are such domain specific languages (DSL), that offer varying support for string interpolation, (strict) typing, functions and validation.
+
+### Infrastructure as Code
+
+A prime example for the application of configuration languages are IaaS^[Infrastructure as a Service] products.
+These solutions offer great flexibility with regard to resource provision (computing, storage, load balancing, etc.), network setup and scaling of (virtual) servers.
+Although the primary interaction with those systems is imperative, maintaining entire applications' or company's environments manually comes with obvious drawbacks.
+
+Changing and undoing changes to existing networks requires intricate knowledge about its topology which in turn has to be meticulously documented.
+Undocumented modification pose a significant risk for *config drift* which is particularly difficult to undo imperatively.
+Beyond that, interacting with a system through its imperative interfaces demands qualified skills of specialized engineers.
+
+The concept of "Infrastructure as Code" (*IaC*) serves the DevOps principles.
+IaC tools help to overcome the need for dedicated teams for *Dev*elopment and *Op*erations by allowing to declaratively specify the dependencies, topology and virtual resources.
+Optimally, different environments for testing, staging and production can be derived from a common base and changes to configurations are atomic.
+As an additional benefit, configuration code is subject to common software engineering tooling;
+It can be statically analyzed, refactored and version controlled to ensure reproducibility.
```

Comment:
```suggestion
The concept of "Infrastructure as Code" (*IaC*) serves the DevOps principles.
IaC tools help to overcome the need for dedicated teams for *Dev*elopment and *Op*erations by allowing to declaratively specify the dependencies, topology and virtual resources.
Optimally, different environments for testing, staging and production can be derived from a common base and changes to configurations are atomic.
As an additional benefit, configuration code is subject to common software engineering tooling;
It can be statically analyzed, refactored and version controlled to ensure reproducibility.
A subset of IaC is focused on largely declarative configuration based on configuration files that are interpreted and "converted" into imperative platform dependent instructions. 
```